server/src/main/java/io/deephaven/server/plugin/python/ObjectTypeAdapter.java

```diff
@@ -41,6 +42,37 @@ ExporterAdapter.class, new ExporterAdapter(exporter),
         out.write(bytes);
     }
 
+    @Override
+    public Kind supportsBidiMessaging(Object object) {
+        return objectTypeAdapter.hasAttribute("create_client_connection") ? Kind.BIDIRECTIONAL
```

Comment:
Do you have python side changes corresponding w/ create_client_connection?

Reply:
No, no point in starting that work until the "are we all okay with breaking old plugin APIs" (vs "how do we handle both cases without making an API that irritates its users") and "is this an appropriate way to be flexible enough in proto, java" points are settled.